defs/bugs/storage/ceph.yaml

```diff
@@ -0,0 +1,18 @@
+38745:
+  requires:
+    and:
+      - apt:
+          ceph-osd:
+            - min: 15.2.6
+              max: 15.2.10
+    or:
+      - property: core.plugins.storage.ceph.CephDaemonConfigShowAllOSDs.bluestore_volume_selection_policy
+        value: [use_some_extra]
+        operator: ne
+      - property: core.plugins.storage.ceph.CephDaemonConfigShowAllOSDs.bluestore_volume_selection_policy
+        value: []
+        operator: ne
```

Comment:
As per documentation, the tag should be op instead of operator
https://github.com/canonical/hotsos/tree/master/defs#requires

Reply:
I see what you mean. I think the sos I tested with doesn't have ceph daemon config info so an empty list was returned, and it was then flagging it because it was checking it as eq instead of ne.. since as you pointed out,  it defaults to eq if op is not found.. I will change it.. thanks!

Reply:
I changed it to op but I still see the issue being flagged in the bugs. The sos I'm testing with does not contain that information,
~~~
root@nikhil-Lenovo-Legion-Y540-15IRH-PG0:/home/nikhil/Downloads/sosreport-juju-6879b7-3-lxd-1-00326782-2022-01-19-ylwpcfg# grep -ri bluestore_volume_selection_policy *
root@nikhil-Lenovo-Legion-Y540-15IRH-PG0:/home/nikhil/Downloads/sosreport-juju-6879b7-3-lxd-1-00326782-2022-01-19-ylwpcfg# 
~~~
So this check,
~~~
38745:
  requires:
    and:
      - apt:
          ceph-osd:
            - min: 15.2.6
              max: 15.2.10
    or:
      - property: core.plugins.storage.ceph.CephDaemonConfigShowAllOSDs.bluestore_volume_selection_policy
        value: [use_some_extra]
        op: ne
      - property: core.plugins.storage.ceph.CephDaemonConfigShowAllOSDs.bluestore_volume_selection_policy
        value: []
        op: ne
  raises:
    message: >-
      Vulnerability to a known bug https://tracker.ceph.com/issues/38745.
      Please set bluestore_volume_selection_policy of all OSDs to use_some_extra
~~~
should ideally not be hit because I'd expect value: [] for property check of core.plugins.storage.ceph.CephDaemonConfigShowAllOSDs.bluestore_volume_selection_policy.. so "ne" should not be triggered.. 
But the bug is still hit, and then I realized that this check is wrong..

~~~
    or:
      - property: core.plugins.storage.ceph.CephDaemonConfigShowAllOSDs.bluestore_volume_selection_policy
        value: [use_some_extra]
        op: ne
      - property: core.plugins.storage.ceph.CephDaemonConfigShowAllOSDs.bluestore_volume_selection_policy
        value: []
        op: ne 
~~~
Since it's "or" the first one is true ..

So I changed the bug check to 

~~~
38745:
  requires:
    and:
      - apt:
          ceph-osd:
            - min: 15.2.6
              max: 15.2.10
      - property: core.plugins.storage.ceph.CephDaemonConfigShowAllOSDs.bluestore_volume_selection_policy
        value: [use_some_extra]
        op: ne
    or:
      and:
        - apt:
            ceph-osd:
              - min: 15.2.6
                max: 15.2.10
        - property: core.plugins.storage.ceph.CephDaemonConfigShowAllOSDs.bluestore_volume_selection_policy
          value: []
          op: ne
~~~

Then I get a proper result, i.e the scenario is hit due to the presense of the spillover string in the ceph status, but the bug is not hit due to the sos not containing the property at all (resulting in empty string)

~~~
  potential-issues:
    CephWarnings:
      - 'This node has Ceph OSDs running on it but is not using cpufreq scaling_governor
        in "performance" mode (actual=ondemand). This is not recommended and can result
        in performance degradation. To fix this you can install cpufrequtils and set
        "GOVERNOR=performance" in /etc/default/cpufrequtils. You will also need to
        disable the ondemand systemd service in order for changes to persist. NOTE:
        requires node reboot to take effect. (origin=storage.auto_scenario_check)'
      - Known ceph bug https://tracker.ceph.com/issues/38745 detected. (origin=storage.auto_scenario_check)
~~~